drivers/accel/adxl313/iio_adxl313.c

```diff
@@ -0,0 +1,1200 @@
+static int adxl313_iio_read_raw(void *dev, char *buf, uint32_t len,
+				const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t ret;
+	int16_t accel_x;
+	int16_t accel_y;
+	int16_t accel_z;
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	switch (channel->type) {
+	case IIO_ACCEL:
+		ret = adxl313_get_raw_xyz(adxl313, &accel_x, &accel_y, &accel_z);
+		if (ret)
+			return ret;
+
+		switch (channel->address) {
+		case ACCEL_AXIS_X:
+			ret = no_os_sign_extend32(accel_x, 12);
+			break;
+		case ACCEL_AXIS_Y:
+			ret = no_os_sign_extend32(accel_y, 12);
+			break;
+		case ACCEL_AXIS_Z:
+			ret = no_os_sign_extend32(accel_z, 12);
+			break;
+
+		default:
+			return -EINVAL;
+		}
+
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &ret);
+
+	default:
+		return -EINVAL;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Handles the read request for scale attribute.
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl313_iio_read_scale(void *dev, char *buf, uint32_t len,
+				  const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t vals[2];
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+	int ret;
+	uint8_t full_res = 0;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	vals[1] = SCALE_FACTOR;
+
+	if (channel->type == IIO_ACCEL) {
+		switch (adxl313->dev_type) {
+		case ID_ADXL312:
+			ret = adxl313_get_full_res_setting(adxl313, &full_res);
+			if (ret)
+				return ret;
+			if (full_res)
+				vals[0] = adxl312_iio_scale_table[0];
+			else
+				vals[0] = adxl312_iio_scale_table[adxl313->range - ADXL313_RANGE_FACTOR];
+			return iio_format_value(buf, len, IIO_VAL_FRACTIONAL, 2, vals);
+		case ID_ADXL313:
+			ret = adxl313_get_full_res_setting(adxl313, &full_res);
+			if (ret)
+				return ret;
+			if (full_res)
+				vals[0] = adxl313_iio_scale_table[0];
+			else
+				vals[0] = adxl313_iio_scale_table[adxl313->range];
+			return iio_format_value(buf, len, IIO_VAL_FRACTIONAL, 2, vals);
+		case ID_ADXL314:
+			vals[0] = adxl314_iio_scale;
+			return iio_format_value(buf, len, IIO_VAL_FRACTIONAL, 2, vals);
+		default:
+			return -EINVAL;
+		}
+	} else {
```

Comment:
no need to use `else` here.  just return directly. (same as you did for `adxl313_iio_write_scale`)